guild.go

```diff
@@ -435,121 +344,74 @@ func (c *Client) ModifyCurrentUserNick(guild objects.Snowflake, params *ModifyCu
-		contentType: JsonContentType,
-		body:        data,
-		reason:      reason,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
 	newNick := &ModifyCurrentUserNickParams{}
-	if err = res.JSON(newNick); err != nil {
-		return nil, err
-	}
+	err = NewRequest().
+		Method(http.MethodPatch).
+		Path(fmt.Sprintf(GuildMemberEditCurrentUserNickFmt, guild)).
+		ContentType(JsonContentType).
+		Body(data).
+		Reason(reason).
+		Expect(http.StatusOK).
+		Bind(newNick).
+		Send(c)
 
-	return newNick, nil
+	return newNick, err
 }
 
 func (c *Client) AddGuildMemberRole(guild, user, role objects.Snowflake, reason string) error {
-	res, err := c.request(&request{
-		method:      http.MethodPut,
-		path:        fmt.Sprintf(GuildMemberRoleFmt, guild, user, role),
-		contentType: JsonContentType,
-		reason:      reason,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-	return nil
+	return NewRequest().
+		Method(http.MethodPut).
+		Path(fmt.Sprintf(GuildMemberRoleFmt, guild, user, role)).
+		ContentType(JsonContentType).
+		Reason(reason).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) RemoveGuildMemberRole(guild, user, role objects.Snowflake, reason string) error {
-	res, err := c.request(&request{
-		method:      http.MethodDelete,
-		path:        fmt.Sprintf(GuildMemberRoleFmt, guild, user, role),
-		contentType: JsonContentType,
-		reason:      reason,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-	return nil
+	return NewRequest().
+		Method(http.MethodDelete).
+		Path(fmt.Sprintf(GuildMemberRoleFmt, guild, user, role)).
+		ContentType(JsonContentType).
+		Reason(reason).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) RemoveGuildMember(guild, user objects.Snowflake, reason string) error {
-	res, err := c.request(&request{
-		method:      http.MethodDelete,
-		path:        fmt.Sprintf(GuildMemberFmt, guild, user),
-		contentType: JsonContentType,
-		reason:      reason,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-	return nil
+	return NewRequest().
+		Method(http.MethodDelete).
+		Path(fmt.Sprintf(GuildMemberFmt, guild, user)).
+		ContentType(JsonContentType).
+		Reason(reason).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) GetGuildBans(guild objects.Snowflake) ([]*objects.Ban, error) {
-	res, err := c.request(&request{
-		method:      http.MethodGet,
-		path:        fmt.Sprintf(GuildBansFmt, guild),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return nil, err
-	}
+	bans := []*objects.Ban{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildBansFmt, guild)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(bans).
```

Comment:
We need to pass a pointer to the slice here, otherwise it will automatically return an error.